hnn_core/cell_response.py

```diff
@@ -476,3 +541,62 @@ def read_spikes(fname, gid_ranges=None):
         cell_response.update_types(gid_ranges)
 
     return cell_response
+
+
+def _read_spikes_hdf5(fname):
+    """Read spiking activity from a hdf5 file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the hdf5 file (e.g., '<pathname>/spk.hdf5') to be read.
+
+    Returns
+    ----------
+    cell_response : CellResponse
+        An instance of the CellResponse object.
+    """
+
+    data = read_hdf5(fname)
+    cell_response = CellResponse(spike_times=data['spike_times'],
+                                 spike_gids=data['spike_gids'],
+                                 spike_types=data['spike_types'])
+
+    if data['times'] == list():
+        data['times'] = np.array(None)
+
+    return cell_response
+
+
+def read_spikes(fname, gid_ranges=None):
+    """Read spiking activity from a hdf5 or txt file.
+
+    Parameters
+    ----------
+    fname : str/Path object
+        Path to the hdf5 or txt file (e.g., '<pathname>/spk.hdf5' or
+        '<pathname>/spk_*.txt') to be read.
+    gid_ranges : dict of lists or range objects | None
+        Dictionary with keys 'evprox1', 'evdist1' etc.
+        containing the range of Cell or input IDs of different
+        cell or input types. If None, data object should contain
+        spike type. (Not applicable for hdf5 files)
+
+    Returns
+    ----------
```

Comment:
nitpick ... the dashes should align with the letters in "Returns" and not overshoot